src/data/set/basic.lean

```diff
@@ -1485,12 +1490,26 @@ by rw range_comp; apply image_subset_range
 lemma range_nonempty_iff_nonempty : (range f).nonempty ↔ nonempty ι :=
 ⟨λ ⟨y, x, hxy⟩, ⟨x⟩, λ ⟨x⟩, ⟨f x, mem_range_self x⟩⟩
 
+lemma image_nonempty_iff_nonempty {f:β → α} {T:set β}:
+  (f '' T).nonempty ↔ T.nonempty  :=
+begin
+  repeat {rw nonempty_def},split;intro A1;cases A1 with a A1,
+  { cases A1 with b A1, apply exists.intro b, apply A1.left},
+  { apply exists.intro (f a), simp, apply exists.intro a, apply and.intro A1, refl},
+end
```

Comment:
```suggestion
lemma image_nonempty_iff_nonempty {f : β → α} {T : set β} :
  (f '' T).nonempty ↔ T.nonempty :=
begin
  split; rintro ⟨a, ha⟩,
  { rcases ha with ⟨b, h, -⟩, exact ⟨b, h⟩ },
  { exact ⟨f a, set.mem_image_of_mem _ ha⟩ },
end
```

Reply:
This lemma is already in mathlib (see my other comment), so would it be ok if I resolve this conversation?